crates/stages/src/test_utils/runner.rs

```diff
@@ -58,18 +57,23 @@ pub(crate) trait ExecuteStageTestRunner: StageTestRunner {
     }
 
     /// Run a hook after [Stage::execute]. Required for Headers & Bodies stages.
-    async fn after_execution(&self, _seed: Self::Seed) -> Result<(), TestRunnerError> {
-        Ok(())
+    fn after_execution(
+        &self,
+        _seed: Self::Seed,
+    ) -> impl Future<Output = Result<(), TestRunnerError>> + Send {
+        async { Ok(()) }
     }
 }
 
-#[async_trait::async_trait]
 pub(crate) trait UnwindStageTestRunner: StageTestRunner {
     /// Validate the unwind
     fn validate_unwind(&self, input: UnwindInput) -> Result<(), TestRunnerError>;
 
     /// Run [Stage::unwind] and return a receiver for the result.
-    async fn unwind(&self, input: UnwindInput) -> Result<UnwindOutput, StageError> {
+    fn unwind(
```

Comment:
since this is the impl of the trait, rust should be able to handle async fn here

Reply:
yes, you're right.  
Looks like `impl Future + Send` is only needed for pub trait, for example, 
https://github.com/paradigmxyz/reth/blob/57136e252cb3169626c1a5409689f0993705e7a7/crates/stages/src/stage.rs#L250-L259
If I use `async` here, the compiler throws a errors below:
```rust
error: use of `async fn` in public traits is discouraged as auto trait bounds cannot be specified
   --> crates/stages/src/stage.rs:253:5
    |
253 |     async fn execute_ready(&mut self, input: ExecInput) -> Result<(), StageError> {
    |     ^^^^^
    |
    = note: you can suppress this lint if you plan to use the trait only in your own code, or do not care about auto traits like `Send` on the `Future`
    = note: `-D async-fn-in-trait` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(async_fn_in_trait)]`
help: you can alternatively desugar to a normal `fn` that returns `impl Future` and add any desired bounds such as `Send`, but these cannot be relaxed without a breaking API change
    |
253 ~     fn execute_ready(&mut self, input: ExecInput) -> impl std::future::Future<Output = Result<(), StageError>> + Send {async {
254 |         poll_fn(|cx| self.poll_execute_ready(cx, input)).await
255 ~     } }
    |


```